src/server/scripts/DragonIsles/AberrusTheShadowedCrucible/boss_kazzara_the_hellforged.cpp

```diff
@@ -15,28 +15,129 @@
+    SPELL_MOLTEN_SCAR_DAMAGE                = 402420,
+    SPELL_OVERRIDE_POWER_DISPLAY            = 402550,
+    SPELL_PERIODIC_ENERGIZE                 = 402538,
+    SPELL_RAY_OF_ANGUISH_AREA               = 402260,
+    SPELL_RAY_OF_ANGUISH_AREATRIGGER        = 402219,
+    SPELL_RAY_OF_ANGUISH_DAMAGE             = 402207,
+    SPELL_RAY_OF_ANGUISH_DAMAGE_AREA        = 407218,
+    SPELL_RAY_OF_ANGUISH_AURA_10_SECONDS    = 409256,
+    SPELL_RAY_OF_ANGUISH_AURA_45_SECONDS    = 411925,
+    SPELL_RAY_OF_ANGUISH_DECREASE_SPEED     = 402299,
+    SPELL_RAY_OF_ANGUISH_FIXATE             = 402253,
+    SPELL_RAY_OF_ANGUISH_MISSILE            = 402187,
+    SPELL_RAYS_OF_ANGUISH                   = 407069,
+    SPELL_RAYS_OF_ANGUISH_RIFTS_CHECK       = 407068,
+    SPELL_RIFTBURN                          = 406530,
+    SPELL_TERROR_CLAWS_DAMAGE               = 406597,
+    SPELL_TERROR_CLAWS_PERIODIC_AURA        = 404743,
+    SPELL_TERROR_CLAWS_PERIODIC_DAMAGE      = 401898,
+    SPELL_TERROR_CLAWS_SELECTOR             = 404744,
+    SPELL_TERROR_CLAWS_SHADOWFLAME_DAMAGE   = 406601,
+    SPELL_WINGS_OF_EXTINCTION               = 403326
+};
+
+enum KazzaraTexts
+{
+    SAY_AGGRO                               = 0,
+    SAY_WINGS_OF_EXTINCTION                 = 1,
+    SAY_ARMOR_80_PCT                        = 2,
+    SAY_RAYS_OF_ANGUISH                     = 3,
+    SAY_HELLBEAM                            = 4,
+    SAY_HELLBEAM_ALERT                      = 5,
+    SAY_TERROR_CLAWS                        = 6,
+    SAY_ARMOR_60_PCT                        = 7,
+    SAY_ARMOR_40_PCT                        = 8,
+    SAY_KILL                                = 9,
+    SAY_DIED                                = 10,
+
+    // Ray of Anguish
+    SAY_FIXATE                              = 0
+};
+
+enum KazzaraEvents
+{
+    EVENT_RAYS_OF_ANGUISH                   = 1,
+    EVENT_TERROR_CLAWS                      = 2,
+    EVENT_WINGS_OF_EXTINCTION               = 3,
+    EVENT_ENERGIZE                          = 4,
+    EVENT_DREAD_RIFT                        = 5,
+
+    // Ray of Anguish
+    EVENT_MOLTEN_SCAR                       = 6
+};
+
+enum KazzaraActions
+{
+    ACTION_HEALTH_BELOW_80_PCT              = 1,
+    ACTION_HEALTH_BELOW_60_PCT              = 2,
+    ACTION_HEALTH_BELOW_40_PCT              = 3
 };
 
 // 201261 - Kazzara the Hellforged
 struct boss_kazzara_the_hellforged : public BossAI
 {
-    boss_kazzara_the_hellforged(Creature* creature) : BossAI(creature, DATA_KAZZARA_THE_HELLFORGED) { }
+    boss_kazzara_the_hellforged(Creature* creature) : BossAI(creature, DATA_KAZZARA_THE_HELLFORGED)
+    {
+        Initialize();
+    }
+
+    void Initialize()
+    {
+        _under80Percent = false;
+        _under60Percent = false;
+        _under40Percent = false;
+        me->SetPower(POWER_ENERGY, 0);
+    }
+
+    void KilledUnit(Unit* victim) override
+    {
+        if (!victim->IsPlayer())
```

Comment:
We can improve this by setting a bool to check whether there's a delay until next. We don't want spammable says upon killing players. @mdX7 texts cooldown wen 👀